### ISSUE TITLE
Automatically link to a default directory

### ISSUE THREAD

**Issue by dbartholomae** [Schniz/fnm#558]
Hi there!

One problem I am running into with fnm is that in my IDE, WebStorm, I need to predefine the path to node as the IDE does not run the fnm env command before checking the PATH env variable. If fnm linked e. g. aliases the current version to a fixed current path like nvm does, I could just enter that path.

**Comment by Schniz** [comment]
I have a solution in mind and I hope I have the time to do it in the near future

**Comment by lcrockett** [comment]
I can second this as it would solve my issue using [evalcache](https://github.com/mroth/evalcache) in conjunction with `fnm`.

**Comment by Schniz** [comment]
Yes. Just need some time! I have a lot of things on my plate these days. :)

**Comment by fellnerse** [comment]
I would love this as well! If it's not too complicated I might be able to help

**Comment by felipesere** [comment]
Any  update on this? I'm stumbling into this too :)

**Comment by tuscen** [comment]
This could also solve fnm integration story with other non-supported shells like nushell as well.

**Comment by nadirvishun** [comment]
still no update, I change to use nvm now....

**Comment by Schniz** [comment]
sorry that I am not helping: it's just not a use case I have (I am using vim, therefore I am using a shell-driven workflow and don't have this issue). This is an open source project and y'all can contribute and share ideas too 😌

anyways:

a simple solution would be to have a file in `~/bin/default-node/node` with the following contents:

```bash
#!/bin/sh
fnm exec --using=default node "$@"
```

then you can set `~/bin/default-node` to be your Node.js directory. You can probably do the same for `npm` (`fnm exec --using=default npm "$@"`)

I tried creating that binary and it's working:

```sh-session
$ cat ~/bin/default-node
#!/bin/sh
fnm exec --using=default node "$@"
$ fnm use 12
Using Node v12.22.9
$ node -v
v12.22.9
$ ~/bin/default-node --version
v16.15.1
```

That is basically a userland implementation of #609, but you can do that while I am not working on this because I am doing other things right now. sorry for making you all wait <picture data-single-emoji=":blob_pray:" title=":blob_pray:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..iA_boWSRD6LiKJyg.stJR3dYGOYaLIPOl24TsPCuS5Qwpyudnx6-Nu1QbAPIlL27h7LTnqA2vMf22HJ0NJqdMjyPEC67cccow7PuS_386IjliSd7a4CJKbq02rHTX3W5PMoLJHLrRKQ.F0HaB81T3ICii5wT5hZrTQ" alt=":blob_pray:" align="absmiddle"></picture>

**Comment by Lippiece** [comment]
@Schniz 
> I tried creating that binary and it's working:

I'm sorry, don't you mean, e.g., `cat bin/default-node/node`? It appears you are calling the directory `default node` (which doesn't work for me) instead of the node script we put there (which works).

Nevertheless, Webstorm doesn't like this path for some reason and can't use it.

<details>
<summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/47634624/229892311-a9b081a4-ff8c-4a4e-813d-96468c9eb67c.png)

</details>

**Comment by nadirvishun** [comment]
in windows：
- if you set env `FNM_DIR` , then in `$FNM_DIR\aliases\default`
- if default, then in `~\AppData\Roaming\fnm\aliases`

**Comment by MaGnaL** [comment]
I took another approach, because using the default version all the time didn't fit my use case. This is currently for windows only, but I guess it can be adapted to linux and macos as well.

I created `node.cmd` with the following content:

```cmd
@ECHO OFF
fnm exec --using .nvmrc node.exe %*
```

I set this file as "Node interpreter" in WebStorm. It passes all of the node calls triggered by WebStorm through that file. Which means it always executes with the node version defined in `.nvmrc`.

Additionally I did the same for the "Package manager" setting using a `npm.cmd` with this content:

```cmd
@ECHO OFF
fnm exec --using .nvmrc npm.cmd %*
```

I guess this can be adapted to yarn, pnpm, etc.

That setup works for me right now. The advantage is, that the only master for the node version is `.nvmrc`, both for WebStorm and the command line.

**Comment by simhnna** [comment]
I wrote a small plugin for jetbrains IDEs. it switches out the node path in the project runtime configuration every time .nvmrc or .node-version files are changed. I'm in the process of getting it published, you can currently download it from [here](https://plugins.jetbrains.com/plugin/32593-fnm-version-switcher)

That way terminals still work as expected, no PATH manipulation or script overrides are needed
